### ISSUE TITLE
Soft launch meta issue

### ISSUE THREAD

**Issue by tmeasday** [meteor/guide#160]
### Both re-read each every article

- Typos / formatting
- Check links work
- Ensure references include links (e.g. “next section”)
- Take Screenshots
- Check code snippets (ensure work w/ 1.2)
- Make a list of non-trivial incomplete parts
  
### Todos app code

- [x] Fix #33 #34 #36 for code style guide
- [x] Using simpleschema#validate in validateNewUser doesn't work with useraccounts:flow routing because of inspecting error details. Add special case for validation error https://github.com/meteor-useraccounts/flow-routing/pull/23 https://github.com/meteor/guide/issues/163 [note: not actually an issue in Todos since we don't need to do this when using useraccounts]
- [x] Backtrack to ordered arguments for publications
- [x] [meteor/todos#57](https://github.com/meteor/todos/issues/57) - simple schema and audit-argument-checks https://github.com/aldeed/meteor-simple-schema/pull/518
- [x] Need to resolve denorm code issue: https://github.com/meteor/todos/pull/60
- [x] Confirm that audit-argument-checks even works with keyword args...
- [x] [meteor/todos#56](https://github.com/meteor/todos/pull/56) - fix re-rendering and triple-function
- [x] Update parts of the code that need i18n with comments referring to the [GitHub issue](https://github.com/meteor/todos/issues/59)

### Content

- [ ] Add section to methods about HTTP API - [ Should we have something about HTTP API/server side routes?    ](https://github.com/meteor/guide/issues/49) - fix simple:rest to allow attaching options to existing method
- [x] Update code snippets with field reactivity: https://github.com/meteor/guide/issues/173
- [x] Address all comments from @splendido review https://github.com/meteor/guide/pull/154
- [x] Denormalizer content got lost between methods and collections (waiting on resolution of PR, https://github.com/meteor/todos/pull/60)
- [x] Backtrack to ordered arguments for publications
- [x] Finalize article order
- [x] Confirm that postCSS content is up to date [meteor/guide#157](https://github.com/meteor/guide/issues/157)
- [x] Build tool package loading reword [meteor/guide#151](https://github.com/meteor/guide/issues/151)
- [x] Fix security section [meteor/guide#148](https://github.com/meteor/guide/issues/148)
- [x] Method call lifecycle improvement [meteor/guide#147](https://github.com/meteor/guide/issues/147)
- [x] Deal with i18n situation[ Figure out tap:i18n section in UX guide](https://github.com/meteor/guide/issues/122)
- [x] Bulk updates [ Recommend bulk updates in migrations, rather than iterating over all items and running a new update query](https://github.com/meteor/guide/issues/118)
- [x] [Add SEO](https://github.com/meteor/guide/issues/115) / prerender.io - dfischer:prerenderio
- [x] Remove references to cursor-utils: [ Build a `cursor-utils` package that allows "slicing" cursors](https://github.com/meteor/guide/issues/73)
- [x] Example of waiting on specific data [meteor/guide#159](https://github.com/meteor/guide/issues/159) 
- [x] Add some text about how to do CD w/ circle (both for apps on Galaxy/mother and packages)
- [x] Update data loading article to filter fields in all publications
- [x] Update security article to show example of `Lists.publicFields`

### Website

- [x] Figure out how to factor images into separate directory
- [x] Deal with Blaze chapter rendering problems
- [x] Order articles [ "Welcome!" probably shouldn't be the 6th article in the sidebar     ](https://github.com/meteor/guide/issues/149)
- [x] Fix or remove search??? - disabled by Sashko
- [x] Fix mobile code snippets [meteor/guide#125](https://github.com/meteor/guide/issues/125) 

### Stretch

- [x] Investigate markdown rendering on Atmosphere... https://atmospherejs.com/alanning/roles is particularly bad - the entire document is one huge broken link.

**Comment by stubailo** [comment]
Goal: proofread every article by the end of today

**Comment by stubailo** [comment]
### Proofreading progress, Sashko

- [ ] Welcome!
- [ ] Collections and Schemas
- [ ] Publications and Data Loading
- [ ] Methods
- [ ] URLs and Routing
- [ ] Users and Accounts
- [ ] User Interfaces
- [ ] Blaze
- [ ] Security
- [ ] Deployment and Monitoring
- [ ] Build system

**Comment by stubailo** [comment]
Collections notes:

- "synchronous fibers-based API" might be super confusing

**Comment by stubailo** [comment]
Data loading notes:

- It sure would be nice to have some kind of diagram or video about publications
- We might not need it immediately, but it might be good to have a section about subscription life cycle just like we have for methods. I don't even know all of the things that happen when you subscribe (live queries being initialized, when the ready state happens, merge box, etc)
- In the place where we recommend loading data inside templates, it's important to mention that you should stop subscriptions when you're not using them, and the easiest way to do that is to tie the subscription life cycle to the template that is using the data. When the user navigates away from that view, the subscription is stopped and the data unloaded, saving valuable server resources!

Stopped at "fetching data"

**Comment by tmeasday** [comment]
Method nodes:

 - The code snippet defining the method here http://guide.meteor.com/methods.html#advanced-boilerplate doesn't actually ever define the method

**Comment by tmeasday** [comment]
Accounts notes:
 - needs a SS of accounts UI
 - throwing errors from `validateNewUser`

**Comment by tmeasday** [comment]
Blaze notes:
 - need to complete example http://guide.meteor.com/blaze.html#composition

**Comment by tmeasday** [comment]
Security notes:
 - XXX: about explaining validated method http://guide.meteor.com/security.html#validated-method
 - XXX: about sikka http://guide.meteor.com/security.html#rate-limiting
 - Text about "passing in a filter" seems weirdly the same as "passing in fields": http://guide.meteor.com/security.html#publication-options

**Comment by tmeasday** [comment]
Deployment notes:
 - Use a screenshot that actually shows deployment w/ multiple containers

**Comment by tmeasday** [comment]
Build system notes:
 - XXX Note about bower
 - NPM 1.2 ?
 - XXX: callbacks from NPM (content is in unmerged package article)

**Comment by stubailo** [comment]
Routing notes:

- What should the `appBody` template be called? `listsShowPage`? `listsShow`? `forceLoggedIn`?
- Should we rename our routes?
- listShowPage code sample needs to be rewritten
- "animation technique" -> should link to the right part of the UX article
- Link to "template as block helper" section in Blaze article
- Link to the right place for analytics, perhaps remove this content since it looks like a duplicate
- Server-side routing should also link to Restivus, which is in many ways a more complete/customizable solution compared to simple:rest.

**Comment by stubailo** [comment]
UI Notes:

- The first paragraph starts with "Regardless of the rendering library that you are using", I think it would be better to start off mentioning Blaze, React, and Angular. In the future we should perhaps link to resources to learning each, and maybe even talking about how to pick one.
- "styleguide" or "style guide"? It feels really weird seeing that as one word.
- The smart component section just shows the HTML of the listsShowPage, which is not the interesting part. We should include an analysis of the data loading, which is in the JS code.
- Could use some comments in both of the the long code snippets in the "pagination controller pattern" section
- Galaxy screenshot has meet-eeyore in the URL

**Comment by stubailo** [comment]
Blaze notes:

- About using data context in template helpers, we say: "This can be hard to reason about, so it’s often a good idea to instead pass the required data into the helper as an argument." - in that case, why do we do it in Todos and in the first example? Perhaps we should change the code.
- Example for passing keyword arguments in the template inclusion section uses a code snippet which no longer exists in the example app
- Propose renaming "helpers in tags" to something like "attribute helpers"
- XXX: we are violating our own rule here and reading into the sub-component
- I'm not a big fan of the name "utility library" - this seems to encourage just having a big "utils.js" file that has all of your complex stuff

**Comment by stubailo** [comment]
Deployment notes

- Modulus section says "see below" for setting up a custom database, but I can't find anything about that in the article
- I feel like we could make Galaxy more prominent - right now it looks like just one of 5 options, we should make it look like the most obvious choice, because it is

**Comment by stubailo** [comment]
Build tool notes

- Temporarily removed note about Bower, should put it back later
- Made a note that NPM will be seamless in 1.3, mentioned meteorhacks:npm for now
- Removed typescript for now, pending Meteor 1.3 and talking to Uri
- Moved the NPM callback content into this article

**Comment by stubailo** [comment]
Data loading notes:

- First code snippet: "In this particular example, the subscription is initialized when the app starts and never stopped" this code sample actually doesn't have a subscription. Perhaps we should add one.
- We don't have a CSS style for "notes" (technically, block quotes...) we should consider adding it soon but it's not a blocker
- "subscribing to data" section has no code snippet
- I think we should rename "Organizing subscriptions" to "subscribe in UI components"
- Reorganized headings, Tom should look over it

**Comment by stubailo** [comment]
Method notes:

- Added a note about future forms article

**Comment by stubailo** [comment]
Notes from call with Tom:

- Sashko should re-read the denormalization section/migration section to see if the “real new content” has been added
- Tom will add a section about subscription life cycle - merge box and whatnot - and get it checked with someone else, probably Glasser
- Also, something about stopping subscriptions
- Let’s think about which diagrams we need for the final release
- Use a callback in the composition example
- Add a point about whitelisting keys on filters passed from the client
- Finish up the security checklist, and figure out what to do about the things that aren’t mentioned anywhere else
- Fix deployment screenshots, black out URL
- Tom should read over new block of content about NPM and callbacks
- New template names:
  - `Lists_show_page`
  - `Lists_show`
  - `App_forceLoggedIn`
  - `App_body`
- Route names
  - `Lists.show`
  - `App.home`
- Some subscription names in routing article are not correct - `lists/public`
- Rename styleguide to style guide
- Make explicit recommendation to avoid data context in helpers
- Sashko to write block quote CSS

## Plan
- Soft launch week of Jan 4?
- Tom to write initial stab at blog post
- Get Tim to tell Sashko how to finish article

**Comment by tmeasday** [comment]
## Build system

 add something about cosmos:browserify for now?

**Comment by stubailo** [comment]
Tasks for Sashko, to discuss during meeting with Zol 1/4/16:

- [ ] Fix naming to match code style guide in Todos app
- [ ] Methods and HTTP API, link to it from server side routing
- [ ] Update code snippets in Guide with field reactivity from Todos app
- [ ] Investigate fix for Atmosphere markdown (Zol?)
- [ ] Remove modulus, move free hosting up, talk to Ryan
- [ ] Go over all of tom's changes since Sashko went on Vacation
- [ ] Re-read denormalization and migrations sections
- [ ] Talk about whitelisting filter keys passed from client
- [ ] Finish security checklist
- [ ] Write block quote/side note CSS
- [ ] Edit blog post
- [ ] What to do about code style article :question:
- [ ] Add something about cosmos:browserify to NPM section? Move out of React guide?

**Comment by stubailo** [comment]
Launched! Blog post going out tomorrow morning.
